cmd/zed/zed.d/statechange-led.sh

```diff
@@ -91,12 +93,14 @@ check_and_set_led()
 state_to_val()
 {
 	state="$1"
-	if [ "$state" = "FAULTED" ] || [ "$state" = "DEGRADED" ] || \
-	   [ "$state" = "UNAVAIL" ] ; then
-		echo 1
-	elif [ "$state" = "ONLINE" ] ; then
-		echo 0
-	fi
+	case "$state" in
+		FAULTED|DEGRADED|UNAVAIL)
+			echo 1
+			;;
+		ONLINE)
```

Comment:
i blindly translated this, but it could be better served as ONLINE => 0, anything else => 1 maybe

Reply:
There are a couple valid states not listed here, (e.g. REMOVED, SPLIT, OFFLINE) so I'm not sure "anything else => 1`" would be quite right either.  A third "we don't care" state might actually be preferable.  But I agree, let's leave that for another time and stick with the direct translation.

Reply:
In that case, may it be preferable to explicitly ignore `[ -z "$val" ]`, instead of the current behaviour which would to write a null string to the `/fault` file five times?